com.ca.lsp.cobol/lsp-core-cobol-parser/src/main/java/com/ca/lsp/core/cobol/preprocessor/sub/util/impl/PreprocessorCleanerServiceImpl.java

```diff
@@ -27,36 +27,67 @@
 import java.util.Scanner;
 
 import static com.ca.lsp.core.cobol.preprocessor.ProcessingConstants.*;
+import static java.util.Optional.ofNullable;
 
 public class PreprocessorCleanerServiceImpl implements PreprocessorCleanerService {
-  private Deque<CobolDocumentContext> contexts = new ArrayDeque<>();
+  private Deque<StringBuilder> contexts = new ArrayDeque<>();
+  private TokenUtils tokenUtils;
 
-  public PreprocessorCleanerServiceImpl() {
-    contexts.push(new CobolDocumentContext());
+  @Inject
+  public PreprocessorCleanerServiceImpl(TokenUtils tokenUtils) {
+    this.tokenUtils = tokenUtils;
+    contexts.push(new StringBuilder());
```

Comment:
Inconsistent injection, one is via constructor, other inline [I saw usage: new PreprocessorCleanerServiceImpl(new TokenUtilsImpl());].  Should be both either via constructor, or both via inline.

Reply:
The injection in the given example is in the unit test, and the best practice is to avoid using dependency injection in tests